docs/c8ql/got-tutorial/graph-traversal.md

```diff
@@ -0,0 +1,320 @@
+- Return the new edge document.
+
+```js
+    FILTER parentId != null AND childId != null
+    INSERT { _from: childId, _to: parentId } INTO ChildOf
+    RETURN NEW
+```
+
+## Traverse to the Parents
+
+Now that edges link character documents (vertices), you have a graph that can query to find out who the parents are of another character. In graph terms, you'll start at a vertex and follow the edges to other vertices with a [graph traversal](../graphs/traversals.md).
+
+```js
+FOR c IN Characters
+    FILTER c.name == "Bran"
+    FOR v IN 1..1 OUTBOUND c ChildOf
+        RETURN v.name
+```
+
+The start vertex is followed by `ChildOf`, which is our edge collection. The example query returns only the name of each parent to keep the result short:
+
+```json
+[
+  "Ned",
+  "Catelyn"
+]
+```
+
+### Traversal Query Explanation
+
+This `FOR` loop doesn't iterate over a collection or an array, it walks the graph and iterates over the connected vertices it finds, with the vertex document assigned to a variable (here: `v`). It can also emit the edges it walked as well as the full path from start to end to [another two variables](../graphs/traversals.md#syntax).
+
+In above query, the traversal is restricted to a minimum and maximum traversal depth of 1 (how many steps to take from the start vertex), and to only follow edges in `OUTBOUND` direction. The edges point from child to parent, and the parent is one step away from the child, thus it returns the parents of the child we start at.
+
+You could also do this in two steps, using the document ID.
+
+1. Run the following code block to return Bran's ID.
+
+    ```js
+    FOR c IN Characters
+        FILTER c.name == "Bran"
+        RETURN c._id
+    ```
+
+1. Use the ID returned in the following code block to return parent names.
+
+    ```js
+    FOR v IN 1..1 OUTBOUND "Characters/2901776" ChildOf
+        RETURN v.name
+    ```
+
+The same result will be returned for Robb, Arya and Sansa as starting point. For Jon Snow, it will only be Ned.
+
+## Traverse to the Children
+
+You can also walk from a parent in reverse edge direction (`INBOUND` that is) to the children:
+
+```js
+FOR c IN Characters
+    FILTER c.name == "Ned"
+    FOR v IN 1..1 INBOUND c ChildOf
+        RETURN v.name
+```
+
+This returns a list of Ned's children:
+
+```json
+[
+  "Robb",
+  "Sansa",
+  "Jon",
+  "Arya",
+  "Bran"
+]
+```
+
+## Traverse to the Grandchildren
+
+The Lannister family has relations that span from parent to grandchild. Let's change the traversal depth to return grandchildren, which means to go exactly two steps:
+
+```js
+FOR c IN Characters
+    FILTER c.name == "Tywin"
+    FOR v IN 2..2 INBOUND c ChildOf
+        RETURN v.name
+```
+
+```json
+[
+  "Joffrey",
+  "Joffrey"
+]
+```
+
+It might be a bit unexpected, that Joffrey is returned twice. However, if you look at the graph visualization, you can see that multiple paths lead from Joffrey (bottom right) to Tywin:
```

Comment:
```suggestion
It might be a bit unexpected that Joffrey is returned twice. However, if you look at the graph visualization, you can see that multiple paths lead from Joffrey (bottom right) to Tywin:
```

Comma not necessary, but I'm fine with leaving it if we want to make it clearer for ESL readers.